GVFS/GVFS.VirtualFileSystemHook/main.cpp

```diff
@@ -51,16 +51,17 @@ int main(int argc, char *argv[])
         success = ReadFromPipe(
             pipeHandle,
             message,
-            sizeof(message),
+            sizeof(message) - 1,
             &bytesRead,
             &lastError);
 
         if (!success)
         {
             break;
         }
-        
+
         messageLength = bytesRead;
+        message[bytesRead] = 0;

```

Comment:
@benpeart can check my thinking here, but the paths are NUL separated so couldn't this create an invalid path by putting a NUL in the middle of a path?   The read from the pipe is not guaranteed to send based on the paths is it?

Reply:
Valid concern - and maybe we should only add the null character if we are about to write the message out to string.

However, the null character is being added outside of the message, and is not part of the `bytesRead` (and hence not part of the actual pipe message)?